### ISSUE TITLE
[Feature request] Extensible searchEngine(s)

### ISSUE THREAD

**Issue by maybe-hello-world** [scambier/obsidian-omnisearch#416]
**Is your feature request related to a problem? Please describe.**
I would love to be able to attach other search targets (not only my vault) to omnisearch, e.g. gDrive docs, so I can look for info in documents shared with me.

**Describe the solution you'd like**
Currently, there exists only one searchEngine which is hardcoded in the plugin. I'd love to be able to attach multiple search engines so the search would go over all of them. Their implementation might be different, i.e. not always implement cache but rely on other services.

**Describe alternatives you've considered**
Use gdrive website ☹️

**Additional context**
I implemented a dummy searchengine for gdrive and it seems like it might work, but currently, the only way for it is to completely rewrite the current engine. If I missed some existing options to attach other engines, I'd appreciate it if you mention them.

**Comment by scambier** [comment]
I'm not sure to understand the request. You want Omnisearch to use additional sources like your gdrive folder?

It's an Obsidian plugin, so it works in the context of Obsidian. Anything more than that would be greatly out of scope.

**Comment by maybe-hello-world** [comment]
Yes, I mean implementing additional possible sources besides just the vault (more like the ability to implement them).
If you feel it doesn't fit into the plugin, this FR can be closed.

**Comment by scambier** [comment]
Ok I understand. That would be too much out of scope for Omnisearch _itself_, but that's a good use-case for another "Text Extractor"-like plugin. The changes in Omnisearch would be minimal (just another set of calls to a 3rd party plugin or application)

For reference, see how it's done from [here](https://github.com/scambier/obsidian-omnisearch/blob/1c98d8b2d7f84eaffb0a1f1871f87d5532e84713/src/repositories/documents-repository.ts#L142-L179) with Text Extractor. Your request would work a bit differently since the files are not in the vault, but ultimately, it would just be feeding data into Omnisearch.
